[BUG] New MFA feature broken

I redeployed using the new 3.4.8 image, and also tried removing it and redeploying fresh. There is no Enable MFA in Settings as per the notes, and when I try to enable MFA via the user icon at the top right, when clicking Enable MFA it says an error occurred. The docker logs show this:

[ERROR] MFA setup error: Error: Auth repository not initialized
    at DatabaseService.updateUserMfaSecretAsync (file:///app/dist/services/database.js:8924:19)
    at file:///app/dist/server/routes/mfaRoutes.js:49:31

--- a/src/server/models/User.ts
+++ b/src/server/models/User.ts
@@ -174,6 +174,21 @@ export class UserModel {
       params.push(input.passwordLocked ? 1 : 0);
     }
 
+    if (input.mfaEnabled !== undefined) {
+      updates.push('mfa_enabled = ?');
+      params.push(input.mfaEnabled ? 1 : 0);
+    }
+
+    if (input.mfaSecret !== undefined) {
+      updates.push('mfa_secret = ?');
+      params.push(input.mfaSecret);
+    }
+
+    if (input.mfaBackupCodes !== undefined) {
+      updates.push('mfa_backup_codes = ?');
+      params.push(input.mfaBackupCodes);
+    }
+
     if (updates.length === 0) {
       return this.findById(id);
     }

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -10784,40 +10784,48 @@ class DatabaseService {
    * Update MFA secret and backup codes for a user.
    */
   async updateUserMfaSecretAsync(userId: number, secret: string, backupCodes: string): Promise<void> {
-    if (!this.authRepo) {
-      throw new Error('Auth repository not initialized');
+    if (this.authRepo) {
+      await this.authRepo.updateUser(userId, { mfaSecret: secret, mfaBackupCodes: backupCodes });
+      return;
     }
-    await this.authRepo.updateUser(userId, { mfaSecret: secret, mfaBackupCodes: backupCodes });
+    // Fallback to sync for SQLite
+    this.userModel.update(userId, { mfaSecret: secret, mfaBackupCodes: backupCodes });
   }
 
   /**
    * Clear MFA data for a user (disable MFA).
    */
   async clearUserMfaAsync(userId: number): Promise<void> {
-    if (!this.authRepo) {
-      throw new Error('Auth repository not initialized');
+    if (this.authRepo) {
+      await this.authRepo.updateUser(userId, { mfaEnabled: false, mfaSecret: null, mfaBackupCodes: null });
+      return;
     }
-    await this.authRepo.updateUser(userId, { mfaEnabled: false, mfaSecret: null, mfaBackupCodes: null });
+    // Fallback to sync for SQLite
+    this.userModel.update(userId, { mfaEnabled: false, mfaSecret: null, mfaBackupCodes: null });
   }
 
   /**
    * Enable MFA for a user (set mfaEnabled to true).
    */
   async enableUserMfaAsync(userId: number): Promise<void> {
-    if (!this.authRepo) {
-      throw new Error('Auth repository not initialized');
+    if (this.authRepo) {
+      await this.authRepo.updateUser(userId, { mfaEnabled: true });
+      return;
     }
-    await this.authRepo.updateUser(userId, { mfaEnabled: true });
+    // Fallback to sync for SQLite
+    this.userModel.update(userId, { mfaEnabled: true });
   }
 
   /**
    * Update backup codes for a user (after one is consumed).
    */
   async consumeBackupCodeAsync(userId: number, remainingCodes: string): Promise<void> {
-    if (!this.authRepo) {
-      throw new Error('Auth repository not initialized');
+    if (this.authRepo) {
+      await this.authRepo.updateUser(userId, { mfaBackupCodes: remainingCodes });
+      return;
     }
-    await this.authRepo.updateUser(userId, { mfaBackupCodes: remainingCodes });
+    // Fallback to sync for SQLite
+    this.userModel.update(userId, { mfaBackupCodes: remainingCodes });
   }
 
   // ============ ASYNC CHANNEL DATABASE METHODS ============

--- a/src/types/auth.ts
+++ b/src/types/auth.ts
@@ -39,6 +39,9 @@ export interface UpdateUserInput {
   displayName?: string;
   isActive?: boolean;
   passwordLocked?: boolean;
+  mfaEnabled?: boolean;
+  mfaSecret?: string | null;
+  mfaBackupCodes?: string | null;
 }
 
 export interface UserSession {

